pennylane/_qubit_device.py

```diff
@@ -101,6 +101,11 @@ def _scatter(indices, array, new_dimensions):
         new_array[indices] = array
         return new_array
 
+    @staticmethod
+    def _const_mul(constant, array):
+        """Data type preserving multiply operation"""
+        return np.multiply(constant, array, dtype=array.dtype)
```

Comment:
Could be better to use `qml.math.multiply` here - not all devices that inherit from `DefaultQubit` are NumPy native. (Haven't tested if `qml.math.multiply` works as expected, though it should :smile:)

Reply:
Nice point! Changed.